--- ./classes/Link.php.orig	2015-04-17 09:46:19.519206473 +0200
+++ ./classes/Link.php	2015-04-17 09:53:51.783984770 +0200
@@ -486,8 +486,8 @@
 
 		$controller = Dispatcher::getInstance()->getController();
 	
-		if (!empty(Context::getContext()->controller->php_self))
-			$controller = Context::getContext()->controller->php_self;
+		if (!empty($context->controller->php_self))
+			$controller = $context->controller->php_self;
 
 		if ($controller == 'product' && isset($params['id_product']))
 			return $this->getProductLink((int)$params['id_product'], null, null, null, (int)$id_lang);
